surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/StrategyComparator.java

```diff
@@ -0,0 +1,31 @@
+package org.arquillian.smart.testing.surefire.provider;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.arquillian.smart.testing.TestSelection;
+
+class StrategyComparator implements Comparator<TestSelection> {
+    private final Map<String, Integer> priorities;
+    private final List<String> strategies;
+
+    StrategyComparator(List<String> strategies) {
+        this.strategies = strategies;
+        this.priorities = priority();
+    }
+
+    private Map<String, Integer> priority() {
+        return strategies.stream()
+            .collect(Collectors.toMap(Function.identity(), s -> (strategies.size() - strategies.indexOf(s)) * 100));
```

Comment:
no you are right. I am missing this corner case. Good catch. I was thinking about this while implementing but couldn't find this case. Thanks for pointing it to me.